myria/test/test_query.py

```diff
@@ -76,7 +76,8 @@ def local_mock(url, request):
 
     elif url.path == '/dataset/user-public/program-adhoc' \
                      '/relation-relation/data':
-        body = str(TUPLES)
+        limit = request.original.params['limit']
+        body = str(TUPLES[:int(limit if limit != '' else len(TUPLES))])
```

Comment:
Why are you testing for the empty string rather than just truthiness as below in `test_relation.py`?